tests/test_json_schema.py

```diff
@@ -2776,37 +2776,6 @@ class Model(BaseModel):
     }
 
 
-def test_path_modify_schema():
-    class MyPath(Path):
-        @classmethod
-        def __get_pydantic_core_schema__(cls, _source_type: Any, handler: GetCoreSchemaHandler) -> CoreSchema:
-            return handler(Path)
```

Comment:
What's the reasoning behind deleting this test?

Reply:
I deleted cause I thought that we didn't want this behaviour. However, I reconsidered, as described below.

This was failing because, with the changed logic, any class that inherits from `os.PathLike` is treated as a known type by `_get_prepare_pydantic_annotations_for_known_type`, and as such ignores any custom (core or json) schema callbacks.

I could change the logic in `path_schema_prepare_pydantic_annotations` to support custom schema, but this would introduce complexity and arguably go against the logic of the other `*prepare*` functions in this file.

Overall, this makes me doubt somewhat this PR. What do you think?

Reply:
> This was failing because, with the changed logic, any class that inherits from os.PathLike is treated as a known type by _get_prepare_pydantic_annotations_for_known_type, and as such ignores any custom (core or json) schema callbacks.

Ah yeah, we'll definitely want to respect the custom callbacks.

> I could change the logic in path_schema_prepare_pydantic_annotations to support custom schema, but this would introduce complexity and arguably go against the logic of the other *prepare* functions in this file.

Yeah, makes sense. I think probably not worth the complexity now, given that you can just add custom logic to a custom path type and have things work ok with the current logic.

> Overall, this makes me doubt somewhat this PR. What do you think?

Hmm, yeah. I think we can close this based on the above. Thanks for the idea, though!

Reply:
Ok, thanks either way for your time Sydney!